ivadomed/models.py

```diff
@@ -425,14 +428,36 @@ def __init__(self, out_channel=1, depth=3, drop_rate=0.4, bn_momentum=0.1,
         self.last_conv = conv(in_channel // 2, out_channel, kernel_size=3, padding=1)
         self.last_film = FiLMlayer(n_metadata, self.out_channel) if film_layers and film_layers[-1] else None
         self.softmax = nn.Softmax(dim=1)
+        
+        ### ATTENTION MODULE ###
+        if self.attention:
+            
+            self.gating = UnetGridGatingSignal2(self.base_n_filter * 2**(self.depth),
+                                                self.base_n_filter * 2**(self.depth-1), kernel_size=(1, 1),
+                                                is_batchnorm=True)
+            for k in range(1,self.depth+1):
+                
+                self.att_path.append(GridAttentionBlock2D(in_channels=self.base_n_filter * 2**(self.depth-k),
+                                                        gating_channels=self.base_n_filter * 2**(self.depth-1),
+                                                        inter_channels=self.base_n_filter * 2**(self.depth-k),
+                                                        sub_sample_factor=2))
+
 
     def forward(self, features, context=None, w_film=None):
         x = features[-1]
+        gating = self.gating(x)
+
+        for i in range(self.depth):
+            if self.attention:
+                y, att = self.att_path[i](features[(self.depth-1)-i],gating)
+                x = self.up_path[2*i](x, y)
+                
+            else:
+                x = self.up_path[2*i](x, features[(self.depth-1)-i])
+            if self.up_path[2*i+1]:
+                x, w_film = self.up_path[2*i+ 1](x, context, w_film)
+                
 
-        for i in reversed(range(self.depth)):
-            x = self.up_path[-(i + 1) * 2](x, features[i])
-            if self.up_path[-(i + 1) * 2 + 1]:
-                x, w_film = self.up_path[-(i + 1) * 2 + 1](x, context, w_film)
```

Comment:
Looks similar to what was implemented before. You are just playing with different indexes. It's okay, but, when not necessary, I would recommend to keep the current way of indexes.